### PR TITLE
`azurerm_redis_cache` : fix incorrect ssl values for `redis_primary_connection_string` and `secondary_connection_string`

### LINKED ISSUE #23490: azurerm_redis_cache - output redis_primary_connection_string ssl=false but port is 6380
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.5.7

### AzureRM Provider Version

3.75.0

### Affected Resource(s)/Data Source(s)

azurerm_redis_cache

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

# NOTE: the Name used for Redis needs to be globally unique
resource "azurerm_redis_cache" "example" {
  name                = "example-cache"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  capacity            = 2
  family              = "C"
  sku_name            = "Standard"
  enable_non_ssl_port = true
  minimum_tls_version = "1.2"

  redis_configuration {
  }
}
```

### Debug Output/Panic Output

```shell
.
```

### Expected Behaviour

```
example.redis.cache.windows.net:6380,password=xxxxxxxxxxxx=,ssl=True,abortConnect=False
```

### Actual Behaviour

```
example.redis.cache.windows.net:6380,password=xxxxxxxxxxxx=,ssl=false,abortConnect=False
```

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by sinbai**
Since the ports of `primary_connection_string` and `secondary_connection_string` are set to `props.SslPort`, the ssl value should always be true, rather than relying on the value of `enable_non_ssl_port `.

Fix #23490

### CODE REVIEW

**Review by catriona-m — CHANGES_REQUESTED** [review]
Thanks for opening this @sinbai. It looks like there is a test failure relating to this that needs to be addressed:

```
------- Stdout: -------
=== RUN   TestAccRedisCache_withoutSSL
=== PAUSE TestAccRedisCache_withoutSSL
=== CONT  TestAccRedisCache_withoutSSL
    testcase.go:113: Step 1/2 error: Check failed: Check 2/3 error: Bad: wrong SSL setting in connection string: primary_connection_string
--- FAIL: TestAccRedisCache_withoutSSL (2048.00s)
FAIL

```

**Comment by Poil** [comment]
Hi,

Thanks,

Perhaps we need two outputs nonssl_primary_connection_string & primary_connection_string to cover every usage ?

of course, nonssl_primary_connection_string should use nonssl port with ssl=false and primary_connection_string should use ssl port with ssl=true ?

**Comment by sinbai** [comment]
> Thanks for opening this @sinbai. It looks like there is a test failure relating to this that needs to be addressed:
> 
> ```
> ------- Stdout: -------
> === RUN   TestAccRedisCache_withoutSSL
> === PAUSE TestAccRedisCache_withoutSSL
> === CONT  TestAccRedisCache_withoutSSL
>     testcase.go:113: Step 1/2 error: Check failed: Check 2/3 error: Bad: wrong SSL setting in connection string: primary_connection_string
> --- FAIL: TestAccRedisCache_withoutSSL (2048.00s)
> FAIL
> ```

Hi @catriona-m thanks for your feedback. The code has been updated. Could you please take another look?

File: internal/services/redis/redis_cache_resource_test.go [hunk 1]
```diff
@@ -1492,7 +1492,7 @@ resource "azurerm_redis_cache" "test" {
 `, data.RandomInteger, data.Locations.Primary, data.RandomInteger)
 }
 
-func testCheckSSLInConnectionString(resourceName string, propertyName string, requireSSL bool) acceptance.TestCheckFunc {
+func testCheckSSLInConnectionString(resourceName string, propertyName string) acceptance.TestCheckFunc {
```
Comment (catriona-m) [inline internal/services/redis/redis_cache_resource_test.go:1495]: given that the connection string is now specifically set to be true now, we probably don't need to have this check in the tests at all anymore?

Comment (sinbai) [reply internal/services/redis/redis_cache_resource_test.go]: Good suggestion, fixed. Thanks.

**Review by catriona-m — APPROVED** [review]
Thanks @sinbai LGTM!

**Pull request merged**